homeassistant/components/twinkly/light.py

```diff
@@ -185,16 +185,47 @@ async def async_turn_on(self, **kwargs: Any) -> None:
                 if isinstance(self._attr_rgbw_color, tuple):
```

Comment:
What could it be that's not a tuple ? You just assigned it and it came from kwargs. We would never pass a non-tuple.

Reply:
Also, you should only assign it to `self._attr_rgbw_color` after the call was a success. Otherwise your locally cached state of the device is wrong.

Reply:
Fixed